reward/staking_manager.go

```diff
@@ -40,65 +41,81 @@ type blockChain interface {
 }
 
 type StakingManager struct {
-	ac           *addressBookConnector
-	sic          *stakingInfoCache
-	gh           governanceHelper
-	bc           blockChain
-	chainHeadCh  chan blockchain.ChainHeadEvent
-	chainHeadSub event.Subscription
-	isActivated  bool
+	addressBookConnector *addressBookConnector
+	stakingInfoCache     *stakingInfoCache
+	stakingInfoDB        *stakingInfoDB
+	governanceHelper     governanceHelper
+	blockchain           blockChain
+	chainHeadChan        chan blockchain.ChainHeadEvent
+	chainHeadSub         event.Subscription
+	isActivated          bool
 }
 
-func NewStakingManager(bc blockChain, gh governanceHelper) *StakingManager {
+func NewStakingManager(bc blockChain, gh governanceHelper, dbm database.DBManager) *StakingManager {
+	sidb := &stakingInfoDB{}
+	if dbm != nil {
+		sidb.db = dbm.GetStakingInfoDB()
```

Comment:
I think it's better to add below functions to `type DBManager interface` and save the DB interface in `stakingInfoDB` of `StakingManger`.

- `PutStakingInfo()`
- `GetStakingInfo()`

Reply:
Please refer below.
```
	WriteGovernanceState(b []byte) error
	ReadGovernanceState() ([]byte, error)
```

Reply:
Review reflected